Expanded MNIST example with colab notebooks (#104)
ludwig-ai/ludwig-docs: docs/examples/mnist.md

dataset.

These interactive notebooks follow the steps of this example:

**TODO: point notebook URL to ludwig-ai/ludwig-docs repo before PR merge**

- Ludwig CLI: [![MNIST Image Classification with Ludwig CLI](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/jimthompson5802/ludwig-docs/blob/expanded-mnist-example-with-colab/docs/examples/mnist_colab_notebooks/MNIST_Classification_with_Ludwig_CLI.ipynb)
- Ludwig Python API: [![MNIST Image Classification with Ludwig API](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/jimthompson5802/ludwig-docs/blob/expanded-mnist-example-with-colab/docs/examples/mnist_colab_notebooks/MNIST_Classification_with_Ludwig_API.ipynb)

## Download the MNIST dataset

[MNIST](http://yann.lecun.com/exdb/mnist/) is a collection of gray-scale images of handwritten digits. This collection is made up of 60,000 images for training and 10,000 images for testing model performance.  Each image is 28 X 28 pixels in gray-scale.

```shell
ludwig datasets download mnist
```

> This command will create a dataset `mnist_dataset.csv` in the current directory.

Reviewer: Maybe also mention that `ludwig datasets download mnist` creates `training/` and `testing/` directories in the current dir with the dataset images.

Author: done

The columns in the dataset are

|column| description |
|------|-------------|
|image_path|file path string for the image|
|label|single digit 0 to 9 indicating what digit is shown in the image|
|split|integer value indicating a training example (0) or test example (2)|

Sample of images with `label`.
![mnist sample images](mnist_colab_notebooks/images/mnist_sample_digits.png)

## Train

> The Ludwig configuration file describes the machine learning task.  There is a vast array of options to control the learning process.  This example only covers a small fraction of the options.  Only the options used in this example are described.  Please refer to the [Configuration Section](../../configuration) for all the details.

Reviewer: Condensed wording.  Just a suggestion, not really necessary.
```suggestion
The Ludwig configuration file describes the machine learning task. This example only uses a small subset of the options provided by Ludwig. Please refer to the [Configuration Section](../../configuration) for all the details.
```

    ```python
    # create Ludwig configuration dictionary
    config = {
      'input_features': [
        {
          'name': 'image_path',
          'type': 'image',
          'preprocessing': {'num_processes': 4},
          'encoder': 'stacked_cnn',
          'conv_layers': [
            {'num_filters': 32, 'filter_size': 3, 'pool_size': 2, 'pool_stride': 2},
            {'num_filters': 64, 'filter_size': 3, 'pool_size': 2, 'pool_stride': 2, 'dropout': 0.4}
          ],
         'fc_layers': [{'output_size': 128, 'dropout': 0.4}]
        }
      ],
      'output_features': [{'name': 'label', 'type': 'category'}],
      'trainer': {'epochs': 5}
    }

    # Constructs Ludwig model from config dictionary
    model = LudwigModel(config, logging_level=logging.INFO)

    ```

Train the model.

=== "cli"

    [`ludwig train` command](../../user_guide/command_line_interface/#train)

    ```shell
    ludwig train \
      --dataset mnist_dataset.csv \
      --config config.yaml
    ```

=== "python"

    [train() method](../../user_guide/api/LudwigModel/#train)

    ```python
    # Trains the model. This cell might take a few minutes.

>     train_stats, preprocessed_data, output_directory = model.train(dataset=train_df)

Author: added write-up for  api download
